### PR TITLE
feat: deploy on mantle and mantle testnet

### PR DESCRIPTION
**Pull request by MantisClone**
## Description of the changes

* Deploy the following contracts on Mantle
  * [RequestDeployer](https://explorer.mantle.xyz/address/0xE99Ab70a5FAE59551544FA326fA048f7B95A24B2) - verified
  * [EthereumProxy](https://explorer.mantle.xyz/address/0x171Ee0881407d4c0C11eA1a2FB7D5b4cdED71e6e) - verified
  * [EthereumFeeProxy](https://explorer.mantle.xyz/address/0xe11BF2fDA23bF0A98365e1A4c04A87C9339e8687) - verified
  * [ERC20Proxy](https://explorer.mantle.xyz/address/0x88Ecc15fDC2985A7926171B938BB2Cd808A5ba40) - verified
  * [ERC20FeeProxy](https://explorer.mantle.xyz/address/0x399F5EE127ce7432E4921a61b8CF52b0af52cbfE) - verified
 
* Deploy the following contracts on Mantle Testnet
  * [RequestDeployer](https://explorer.testnet.mantle.xyz/address/0xE99Ab70a5FAE59551544FA326fA048f7B95A24B2) - verified
  * [EthereumProxy](https://explorer.testnet.mantle.xyz/address/0x171Ee0881407d4c0C11eA1a2FB7D5b4cdED71e6e) - NOT verified
  * [EthereumFeeProxy](https://explorer.testnet.mantle.xyz/address/0xe11BF2fDA23bF0A98365e1A4c04A87C9339e8687) - NOT verified
  * [ERC20Proxy](https://explorer.testnet.mantle.xyz/address/0x88Ecc15fDC2985A7926171B938BB2Cd808A5ba40) - NOT verified
  * [ERC20FeeProxy](https://explorer.testnet.mantle.xyz/address/0x399F5EE127ce7432E4921a61b8CF52b0af52cbfE) - NOT verified

* Update `hardhat-etherscan` to latest version 3.1.7
* Add function for verifying RequestDeployer

### CODE REVIEW

File: packages/currency/src/native.ts [hunk 1]
```diff
@@ -123,6 +123,18 @@ export const nativeCurrencies: Record<RequestLogicTypes.CURRENCY.ETH, NativeEthC
...
       network: 'tombchain',
     },
+    {
+      symbol: 'MNT',
+      decimals: 18,
+      name: 'Mantle',
+      network: 'mantle',
+    },
+    {
+      symbol: 'MNT',
```
Comment (benjlevesque) [inline packages/currency/src/native.ts:133]: I think symbols must be unique, and should not be confused with their mainnet counterpart. 
Check what we do for other testnets

File: packages/smart-contracts/scripts-create2/utils.ts
Comment (benjlevesque) [inline packages/smart-contracts/scripts-create2/utils.ts:1]: did you mean to version this?

File: packages/smart-contracts/scripts-create2/deploy-request-deployer.ts [hunk 2]
```diff
@@ -29,3 +29,20 @@ export async function deployDeployer(hre: HardhatRuntimeEnvironment): Promise<vo
     console.error(e);
   }
 }
+
+// Verifies the RequestDeployer (useful if the verification failed after deployment)
+export async function verifyDeployer(hre: HardhatRuntimeEnvironment): Promise<void> {
+  try {
+    const RequestDeployer = await hre.ethers.getContractAt(
+      'RequestDeployer',
+      '0xE99Ab70a5FAE59551544FA326fA048f7B95A24B2',
```
Comment (benjlevesque) [inline packages/smart-contracts/scripts-create2/deploy-request-deployer.ts:38]: what is this address?

**Review by benjlevesque — APPROVED** [review]

File: packages/smart-contracts/scripts-create2/deploy-request-deployer.ts (hunk 2 above)
Comment (leoslr) [reply packages/smart-contracts/scripts-create2/deploy-request-deployer.ts:38]: It's the address of the `RequestDeployer` contract, same on all networks

**Review by leoslr — APPROVED** [review]

File: packages/smart-contracts/scripts-create2/utils.ts
Comment (MantisClone) [reply packages/smart-contracts/scripts-create2/utils.ts:1]: No, I don't think this file needs to be versioned. We edit the `create2ContractDeploymentList` based on which contracts we're trying to deploy.

If you're asking about "create2" in the filename, that's referring to the CREATE2 opcode which is one of the key ingredients to the xdeployer which makes our smart contract addresses the same across multiple chains.

File: packages/smart-contracts/scripts-create2/deploy-request-deployer.ts (hunk 2 above)
Comment (benjlevesque) [reply packages/smart-contracts/scripts-create2/deploy-request-deployer.ts:38]: it should be a variable or should have a comment

Comment (leoslr) [reply packages/smart-contracts/scripts-create2/deploy-request-deployer.ts:38]: There is one in the hardhat config file actually. Maybe we should put it elsewhere and export it

Comment (MantisClone) [reply packages/smart-contracts/scripts-create2/deploy-request-deployer.ts:38]: Good point. Okay I'll fix this.

**Pull request merged**